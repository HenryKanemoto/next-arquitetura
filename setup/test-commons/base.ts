import type {Config} from '@jest/types';
import path from 'path';

// Sync object
const config: Config.InitialOptions = {
    testEnvironment: 'jsdom',
    verbose: true,
    transform: {
        '\\.[jt]sx?$': ['babel-jest', {
            configFile: path.resolve(__dirname, '.babelrc'),
        }],
    },
    setupFilesAfterEnv: [
        path.resolve(__dirname, 'src', 'setupTests.js'),
    ],
    preset: 'ts-jest',
    moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
    testMatch: ['**/?(*.)+(spec|test).ts?(x)']
};

export default config;
