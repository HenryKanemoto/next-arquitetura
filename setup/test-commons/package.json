{
    "name": "@alura/test-commons",
    "version": "0.0.0",
    "private": true,
    "main": "index.js",
    "files": [
        "base.ts"
    ],
    "devDependencies": {
        "jest-styled-components": "7.0.8",
        "@babel/core": "7.22.0",
        "@babel/preset-env": "7.22.0",
        "@babel/preset-react": "7.16.7",
        "@babel/preset-typescript": "7.16.7",
        "@testing-library/dom": "^10.4.0",
        "@testing-library/react": "^13.2.0",
        "@types/react": "^18.3.5",
        "@types/react-dom": "^18.3.0",
        "babel-jest": "28.0.3",
        "jest": "27.0.0",
        "ts-node": "10.7.0"
    },
    "scripts": {
        "test": "echo 'Not implemented'",
        "lint": "echo 'Not implemented'"
    }
}
