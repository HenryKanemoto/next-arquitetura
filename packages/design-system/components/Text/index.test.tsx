import { describe } from "node:test";
import Text from ".";
import { customRender } from "@alura/test-commons/react-testing-library";

const render = customRender()

describe("<Text />", () => {
    test("Text renderiza como h1 quando solicitado", () => {
        const { container } = render(<Text tag="h1">Sample text</Text>);
        expect(container).toMatchSnapshot()
    })
})