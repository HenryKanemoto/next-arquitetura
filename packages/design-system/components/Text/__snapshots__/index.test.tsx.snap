// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renderiza como h1 quando solicitado 1`] = `
<div>
  <h1
    class="sc-blHHSb iPvGx"
  >
    Sample text
  </h1>
</div>
`;
